src/components/Checkbox/Checkbox.tsx

```diff
@@ -1,37 +1,15 @@
 import React from 'react';
-import clsx from 'clsx';
-import {useCheckbox} from '@react-aria/checkbox';
-import {useToggleState} from '@react-stately/toggle';
-import {useFocusRing} from '@react-aria/focus';
 
 import {CheckboxProps} from './Checkbox.types';
-import './Checkbox.scss';
-
-export const Checkbox: React.FC<CheckboxProps> = props => {
-    const {
-        className,
-        isIndeterminate = false
-    } = props;
-    const inputRef = React.useRef<HTMLInputElement>(null);
-    const {inputProps} = useCheckbox(props, useToggleState(props), inputRef);
-    const {isFocusVisible, focusProps} = useFocusRing();
-
-    return (
-        <div className={clsx('moonstone-checkbox', className)}>
-            <input
-                className={clsx('moonstone-checkbox_input', {'hasFocus': isFocusVisible})}
-                ref={inputRef}
-                {...inputProps}
-                {...focusProps}
-            />
-            <svg className={clsx('moonstone-checkbox_icon')} viewBox="0 0 21 21">
-                { isIndeterminate
-                    ? <path d="M4.5 10.5L16.5 10.5" strokeLinecap="round"/>
-                    : <path d="M5 10.75L8.5 14.25L16 6" strokeLinecap="round"/>
-                }
-            </svg>
-        </div>
-    );
+import {UncontrolledCheckbox} from './UncontrolledCheckbox';
+import {ControlledCheckbox} from './ControlledCheckbox';
+
+export const Checkbox: React.FC<CheckboxProps> = ({variant = 'controlled', ...props}) => {
+    // Controlled is the default variant of checkbox to use because components should
+    // be controlled whenever possible
+    return variant === 'uncontrolled'
```

Comment:
It seems we use `variant` for differentiating style, I don't like to use it for technical purposes here. Do you think we can implement it in a transparent way as we did for the `TreeView` component?

Reply:
I'm not sure what property we can check against to differentiate between to use the controlled vs uncontrolled version. Any of the properties could be present in either version, i.e., `checked`, `aria-checked`, etc. Do you think it could be a different prop name? Maybe something like `type`?

Reply:
I'm changing this to be a boolean, `isUncontrolled`, where it is false by default.